scripts/lint-style.py

```diff
@@ -160,6 +161,25 @@ def line_endings_check(lines, path):
         newlines.append((line_nr, line))
     return errors, newlines
 
+def four_spaces_in_second_line(lines, path):
+    errors = []
+    # We never alter the first line, as it does not occur as next_line in the iteration over the
+    # zipped lines below, hence we add it here
+    newlines = [lines[0]]
+    for (_, line), (next_line_nr, next_line) in zip(lines, lines[1:]):
+        modified_next_line = next_line
+        # Check if the current line matches "(lemma|theorem) .* :"
+        if re.search(r"^(protected )?(def|lemma|theorem) .* :$", line):
```

Comment:
This will only work for statements where the keyword (such as `lemma`) is on the same line as the `:`. Would it make sense to try to generalize this?

Reply:
Good idea. I have added that. That only has the slight problem that if we have a doc string like this
```
blba bla
lemma blabla 
bla bla
```
it will get triggered. I'm not sure if this can be avoided without keeping track if we are currently in a doc string by parsing `/--` and `-/`

Reply:
We already have some logic in the style linters for annotating comment lines with `annotate_comments` this sticks a `True` on the end if we are in a comment (or if there is a comment at the end of the line iirc). So like some of the others I think it would be fairly easy to skip linting those lines

Reply:
ah good, I hadn't seen that and now added parsing if we are in comments..., but I will convert it to use `annotate_comments`...

Reply:
Funnily there are some examples of source code inside comments where linting would be a good idea, like one could consider stuff inside ` ``` ...``` ` as code again. But perhaps this is more easily done manually

Reply:
Oh definitely! There are even `begin ... end` blocks inside comments still. But yeah it does seem more complexity than its worth at this point

Reply:
ok, I think this works now: Additional fixes from that here: #7328